src/generator/helpers/constraints.ts

```diff
@@ -0,0 +1,81 @@
+import { DMMF } from '@prisma/client/runtime'
+
+/**
+ * Find the unique identifiers necessary to indentify a field
+ *
+ * Unique fields for a model can be one of (in this order):
+ * 1. One (and only one) field with an @id annotation
+ * 2. Multiple fields with @@id clause
+ * 3. One (and only one) field with a @unique annotation (if there are multiple, use the first one)
+ * 4. Multiple fields with a @@unique clause
+ */
+export function resolveUniqueIdentifiers(model: DMMF.Model): string[] {
```

Comment:
I like this approach of separating the end goal problem into smaller sub problems, LGTM.

Reply:
This comes from the old plugin :)